src/IO/VarInt.h

```diff
@@ -186,6 +188,7 @@ inline const char * readVarUInt(UInt64 & x, const char * istr, size_t size)
 
 inline void writeVarUInt(UInt64 x, WriteBuffer & ostr)
 {
+    chassert(x <= VAR_UINT_MAX);
```

Comment:
I am totally good with the sanity check (thanks). On the other hand, I don't find a principal reason why values > VAR_UNIT_MAX cannot be encoded. It looks to me the only reason for the restriction is that `writeVarUInt*` currently proceeds in 9 chunks à 7 bits (= 63 bits). Special handling for the input's MSB would resolve this limitation. Perhaps add at least TODO comment?

Reply:
>It looks to me the only reason for the restriction is that writeVarUInt* currently proceeds in 9 chunks à 7 bits (= 63 bits)

The problem is, that it will not be backward compatible, so you cannot simply change the implementation of this functions, since older server will not be able to read, and debugging such things can be a nightmare...

Also some clients may rely on this too:
- python driver and it has not such restrictions - it has `while True` - https://github.com/mymarilyn/clickhouse-driver/blob/2942ae34e762eea106b5ab08538d078d6b2b79ef/clickhouse_driver/varint.pyx#L14
- Rust driver has explicit overflow check - https://github.com/suharev7/clickhouse-rs/blob/ecf28f46774773f39c74ee5213ad1e3ea240739b/src/binary/read_ex.rs#L87

Reply:
Uh, I see. So if one ever wanted to get this right, one would need another set of var-int encoding-metthods next to the existing ones (= rather ugly) or add a switch as function parameter (also ugly) :-(

Anyways, a comment above `VAR_UINT_MAX` mentioning
- the restriction is not because of var-int encoding but due to the implementation and
- due to backward compat this cannot be fixed easily
will be helpful for future readers.